### PR TITLE
Finish the span only when the entity is completely drained

### PR DESCRIPTION
**Pull request by Falmarri**
Also, make the 404 operation re-name configurable

### CODE REVIEW

File: kamon-akka-http/src/main/resources/reference.conf [hunk 1]
```diff
@@ -17,9 +17,16 @@ kamon.akka-http {
...
+  not-found-operation-name = "unhandled"
+
   modules {
     kamon-akka-http {
       requires-aspectj = yes
     }
   }
+}
+
+akka {
```
Comment (ivantopo) [inline kamon-akka-http/src/main/resources/reference.conf]: please remove this section.

File: kamon-akka-http/src/main/scala/kamon/akka/http/AkkaHttpMetrics.scala [hunk 2]
```diff
@@ -34,4 +34,8 @@ object AkkaHttpMetrics {
     *    - port: Listening port
     */
   val OpenConnections = Kamon.rangeSampler("akka.http.server.open-connections")
+
+  val RequestEntityTiming = Kamon.timer("akka.http.server.request.entity.read-time")
```
Comment (ivantopo) [inline kamon-akka-http/src/main/scala/kamon/akka/http/AkkaHttpMetrics.scala]: These two metrics should be removed for now if they are not used anywhere. I remember that we talked about having a way to measure the streaming part of the request/response so maybe we can take a look at that in a separate PR!

File: build.sbt [hunk 3]
```diff
@@ -22,13 +22,13 @@ val kamonAkka25         = "io.kamon" %% "kamon-akka-2.5"        % "1.0.0"
 val akkaHttpJson        = "de.heikoseeberger" %% "akka-http-json4s" % "1.18.1"
 val json4sNative        = "org.json4s" %% "json4s-native" % "3.5.3"
 
-val http24         = "com.typesafe.akka" %% "akka-http"          % "10.0.11"
+val http24         = "com.typesafe.akka" %% "akka-http"          % "10.1.1"
```
Comment (ivantopo) [inline build.sbt]: Is this code in any way incompatible with Akka HTTP 10.0.11?

**Review by ivantopo — CHANGES_REQUESTED** [review]

File: kamon-akka-http/src/test/scala/kamon/testkit/TestWebServer.scala [hunk 4]
```diff
@@ -86,6 +91,21 @@ trait TestWebServer extends TracingDirectives {
...
+        path(stream) {
+          complete{
+
+            val longStringContentStream = Source(
+              Range(1, 16).map{ i =>
+              ByteString(
+                100 * ('a' + i).toChar
+              )
+              }
+            )//.initialDelay(1.second)
```
Comment (dpsoft) [inline kamon-akka-http/src/test/scala/kamon/testkit/TestWebServer.scala]: please remove the unused code.

**Pull request merged**